src/configuration/configuration.ts

```diff
@@ -57,6 +57,28 @@ class ConfigurationClass {
         this[option] = vimOptionValue;
       }
     }
+
+    // Get the cursor type from vscode
+    const cursorStyleString = vscode.workspace.getConfiguration().get("editor.cursorStyle") as string;
+    this.userCursor = this.cursorStyleFromString(cursorStyleString);
+  }
+
+  private cursorStyleFromString(cursorStyle: string): vscode.TextEditorCursorStyle {
+    if (cursorStyle === 'line') {
```

Comment:
We should also refine their type `string` to be a union type amongst all the different cursor types... Haha, but you dont have to do that, we don't get much of an advantage from it unfortunately.